app/controllers/organisation/organisation_about_controller.rb

```diff
@@ -1,30 +1,159 @@
+require 'ideal_postcodes'
+require 'pry'
+
 class Organisation::OrganisationAboutController < ApplicationController
   include OrganisationHelper
-  before_action :authenticate_user!, :set_organisation
+  before_action :set_api_key, :authenticate_user!, :set_organisation
+
+  # Renders the initial postcode lookup view
+  def show_postcode_lookup
+    render :postcode_lookup
+  end
 
+  # Renders the full organisation name and address view
   def show
     render :about
   end
 
+  # Makes a request to the IdealPostcodes API to retrieve a list of addresses
+  # matching the given postcode.
+  # If no postcode or an invalid postcode is found, this method redirects the user
+  # back to the :postcode_lookup view.
+  def display_address_search_results
+
+    if !params['postcode']['lookup'].present?
```

Comment:
```suggestion
    unless params['postcode']['lookup'].present?
```
we can use this syntax instead.